dvc/dependency/repo.py

```diff
@@ -46,13 +47,22 @@ def _make_repo(self, **overrides):
             yield repo
 
     def status(self):
-        with self._make_repo() as repo:
-            current = repo.find_out_by_relpath(self.def_path).info
-
-        with self._make_repo(rev_lock=None) as repo:
-            updated = repo.find_out_by_relpath(self.def_path).info
-
-        if current != updated:
+        with self._make_repo() as repo, self._make_repo(
+            rev_lock=None
+        ) as updated_repo:
+            try:
+                current = repo.find_out_by_relpath(self.def_path).info
+                updated = updated_repo.find_out_by_relpath(self.def_path).info
+
+                has_changed = current != updated
+            except OutputNotFoundError:
+                current_path = os.path.join(repo.root_dir, self.def_path)
+                updated_path = os.path.join(
+                    updated_repo.root_dir, self.def_path
+                )
+                has_changed = not filecmp.cmp(current_path, updated_path)
```

Comment:
There is a gotcha in filecmp.cmp:
```suggestion
                has_changed = not filecmp.cmp(current_path, updated_path, shallow=False)
```
without it, it will only check sizes, I think.

Btw, these paths might be dirs and IIRC cmp() doesn't work with dirs. Maybe let's use md5 that we use for cached outputs? See `RemoteLOCAL.get_checksum()`, which you could invoke through `self.repo.cache.local.get_checksum()`.

Reply:
Thanks for pointing out where I can find a checksum function :) I already suspected filecmp wasn't good for this.